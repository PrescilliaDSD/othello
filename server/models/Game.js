const mongoose = require('mongoose')

const { Schema } = mongoose

const GameSchema = new Schema({
  board: {
    type: Array,
    required: true
  },
  player1: {
    pseudo: {
      type: String,
      required: true
    },
    color: {
      type: String,
      required: true
    },
    score: {
      type: Number,
      required: true
    }
  },
  player2: {
    pseudo: {
      type: String
    },
    color: {
      type: String
    },
    score: {
      type: Number
    }
  },
  turn: {
    type: 'String',
    required: true
  },
  status: {
    type: String
  },
  lastPass: {
    type: String
  }
})

module.exports = mongoose.model('Game', GameSchema)
