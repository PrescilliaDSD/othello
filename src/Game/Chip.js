import React from 'react'
import PropTypes from 'prop-types'

const Chip = ({ color }) => {
  return (
    <div className={`${color}-chip chip`}></div>
  )
}

Chip.propTypes = {
  color: PropTypes.string.isRequired
}

export default Chip
