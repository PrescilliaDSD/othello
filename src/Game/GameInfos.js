import React from 'react'
import PropTypes from 'prop-types'

const GameInfos = ({ currentPlayer }) => {
  return (
    <aside className="aside">
      <div className={`aside__chip ${currentPlayer}-chip chip`}></div>
      <button type="button" className="pass-button">passer</button>
    </aside>
  )
}

GameInfos.propTypes = {
  currentPlayer: PropTypes.string.isRequired
}

export default GameInfos
