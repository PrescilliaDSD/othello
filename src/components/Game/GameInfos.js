import React from 'react'
import PropTypes from 'prop-types'

const GameInfos = ({ currentPlayer, setCurrentPlayer, opponent, setOpponent, board, lastPlayerPass, setLastPlayerPass, whiteScore, blackScore, endGame, endMsg, player1, player2 }) => {
  const userShouldPass = () => {
    if (board.every((row) => {
      return row.every((column) => {
        return column !== 'available'
      })
    })) {
      if (lastPlayerPass === opponent || lastPlayerPass === currentPlayer) {
        endGame()
      } else {
        setLastPlayerPass(currentPlayer)
        setCurrentPlayer(opponent)
        setOpponent(currentPlayer)
      }
    }
  }

  return (
    <aside className="aside">
      <div className="msg">{endMsg}</div>
      <p>Score : </p>
      <ul className="score__list">
        <li className="score__list__item">{player1} <div className='aside__chip black-chip chip'><p>{blackScore}</p></div></li>
        <li className="score__list__item">{player2} <div className='aside__chip white-chip chip'><p>{whiteScore}</p></div></li>
      </ul>
      <div className="turn">C&apos;est au tour de : <div className={`aside__chip ${currentPlayer}-chip chip`}></div></div>
      <button onClick={userShouldPass} type="button" className="pass-button">passer</button>
    </aside>
  )
}

GameInfos.propTypes = {
  // string
  currentPlayer: PropTypes.string.isRequired,
  opponent: PropTypes.string.isRequired,
  endMsg: PropTypes.string.isRequired,
  player1: PropTypes.string.isRequired,
  player2: PropTypes.string.isRequired,
  lastPlayerPass: PropTypes.string.isRequired,

  // func
  setCurrentPlayer: PropTypes.func.isRequired,
  setOpponent: PropTypes.func.isRequired,
  endGame: PropTypes.func.isRequired,
  setLastPlayerPass: PropTypes.func.isRequired,

  // array
  board: PropTypes.array.isRequired,

  // number
  whiteScore: PropTypes.number.isRequired,
  blackScore: PropTypes.number.isRequired
}

export default GameInfos
