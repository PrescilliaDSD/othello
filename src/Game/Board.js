import React, { useState, useLayoutEffect } from 'react'
import PropTypes from 'prop-types'

import Chip from './Chip'

const Board = ({ currentPlayer }) => {
  const [board, setBoard] = useState([
    ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
    ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
    ['no', 'no', 'no', 'available', 'no', 'no', 'no', 'no'],
    ['no', 'no', 'available', 'white', 'black', 'no', 'no', 'no'],
    ['no', 'no', 'no', 'black', 'white', 'available', 'no', 'no'],
    ['no', 'no', 'no', 'no', 'available', 'no', 'no', 'no'],
    ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
    ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no']
  ])

  const addChip = async (event) => {
    const coordinates = await event.target.id.split('')
    const newBoard = await board.map((row, i) => {
      return row.map((column, j) => {
        if (i === coordinates[0] - 1 && j === coordinates[1] - 1) {
          return 'black'
        }
        return column
      })
    })
    await parseTopToBottom(newBoard)
  }

  const parseTopToBottom = (boardUpdated) => {
    const boardParsed = boardUpdated.map((row, i) => {
      if (i <= 5) {
        return row.map((column, j) => {
          if (column === 'white') {
            return 'white'
          } else if (column === 'black') {
            return 'black'
          } else if (boardUpdated[i + 1][j] === 'no' || boardUpdated[i + 1][j] === 'black' || boardUpdated[i + 1][j] === 'available') {
            return column
          } else if (currentPlayer === 'black' && boardUpdated[i + 1][j] === 'white' && boardUpdated[i + 2][j] === 'black') {
            return 'available'
          }

        })
      } else {
        return row
      }
    })
    setBoard(boardParsed)
  }

  return (
    <>
      <article className="board">
        <ul className="board__row">
          <li className="row-number"></li>
          <li className="row-number">1</li>
          <li className="row-number">2</li>
          <li className="row-number">3</li>
          <li className="row-number">4</li>
          <li className="row-number">5</li>
          <li className="row-number">6</li>
          <li className="row-number">7</li>
          <li className="row-number">8</li>
        </ul>
        {
          board.map((row, i) => (
            <ul key={`row${i}`} className="board__row">
              <li className="column-number">{ i + 1 }</li>
              {
                row.map((column, j) => {
                  if (column === 'available') {
                    return (
                      <li onClick={addChip} className="board__square clickable" key={`row${i}column${j}`} id={`${i + 1}${j + 1}`}>
                        <Chip color={column} />
                      </li>
                    )
                  }
                  return (
                    <li className="board__square" key={`row${i}column${j}`} id={`${i + 1}${j + 1}`}>
                      <Chip color={column} />
                    </li>
                  )
                })
              }
            </ul>
          ))
        }
      </article>
    </>
  )
}

Board.propTypes = {
  currentPlayer: PropTypes.string.isRequired
}

export default Board
