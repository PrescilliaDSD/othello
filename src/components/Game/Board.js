/* eslint-disable react-hooks/exhaustive-deps */
import React, { useEffect, useRef } from 'react'
import { useParams } from 'react-router-dom'
import PropTypes from 'prop-types'

import Chip from './Chip'

const Board = ({ currentPlayer, setCurrentPlayer, opponent, setOpponent, board, setBoard, setLastPlayerPass, setWhiteScore, setBlackScore, endGame, player1, player2, status, whiteScore, blackScore, lastPlayerPass }) => {
  const id = useParams()
  const isInitialMount = useRef(true)

  const addChip = (i, j) => () => {
    setLastPlayerPass('none')
    const newBoard = board.map((row, k) => {
      return row.map((column, l) => {
        if (k === i && l === j) {
          return currentPlayer
        }
        return column
      })
    })
    if (currentPlayer === 'black') {
      setCurrentPlayer('white')
      setOpponent('black')
    } else {
      setCurrentPlayer('black')
      setOpponent('white')
    }
    setBoard(newBoard)
    parseBoardToTurnChip(newBoard, i, j)
  }

  const isAvailable = (arr, i, j, indI, indJ) => {
    let k = 1
    if (indI === '-' && indJ === 'none') {
      while (i - k >= 0 && arr[i - k][j] === opponent) {
        k += 1
      }
      if (i - k >= 0 && arr[i - k][j] === currentPlayer) {
        return true
      }
    } else if (indI === '+' && indJ === 'none') {
      while (i + k <= 7 && arr[i + k][j] === opponent) {
        k += 1
      }
      if (i + k <= 7 && arr[i + k][j] === currentPlayer) {
        return true
      }
    } else if (indI === 'none' && indJ === '+') {
      while (j + k <= 7 && arr[i][j + k] === opponent) {
        k += 1
      }
      if (j + k <= 7 && arr[i][j + k] === currentPlayer) {
        return true
      }
    } else if (indI === 'none' && indJ === '-') {
      while (j - k >= 0 && arr[i][j - k] === opponent) {
        k += 1
      }
      if (j - k >= 0 && arr[i][j - k] === currentPlayer) {
        return true
      }
    } else if (indI === '+' && indJ === '+') {
      while (i + k <= 7 && j + k <= 7 && arr[i + k][j + k] === opponent) {
        k += 1
      }
      if (i + k <= 7 && j + k <= 7 && arr[i + k][j + k] === currentPlayer) {
        return true
      }
    } else if (indI === '+' && indJ === '-') {
      while (i + k <= 7 && j - k >= 0 && arr[i + k][j - k] === opponent) {
        k += 1
      }
      if (i + k <= 7 && j - k >= 0 && arr[i + k][j - k] === currentPlayer) {
        return true
      }
    } else if (indI === '-' && indJ === '+') {
      while (i - k >= 0 && j + k <= 7 && arr[i - k][j + k] === opponent) {
        k += 1
      }
      if (i - k >= 0 && j + k <= 7 && arr[i - k][j + k] === currentPlayer) {
        return true
      }
    } else if (indI === '-' && indJ === '-') {
      while (i - k >= 0 && j - k >= 0 && arr[i - k][j - k] === opponent) {
        k += 1
      }
      if (i - k >= 0 && j - k >= 0 && arr[i - k][j - k] === currentPlayer) {
        return true
      }
    }
    return false
  }

  const isTurning = (arr, i, j, indI, indJ) => {
    let k = 1
    if (indI === '-' && indJ === 'none') {
      while (i - k >= 0 && arr[i - k][j] === opponent) {
        k += 1
      }
      if (i - k >= 0 && arr[i - k][j] === currentPlayer) {
        for (let index = 1; index < k; index += 1) {
          arr[i - index][j] = currentPlayer
        }
      }
    } else if (indI === '+' && indJ === 'none') {
      while (i + k <= 7 && arr[i + k][j] === opponent) {
        k += 1
      }
      if (i + k <= 7 && arr[i + k][j] === currentPlayer) {
        for (let index = 1; index < k; index += 1) {
          arr[i + index][j] = currentPlayer
        }
      }
    } else if (indI === 'none' && indJ === '+') {
      while (j + k <= 7 && arr[i][j + k] === opponent) {
        k += 1
      }
      if (j + k <= 7 && arr[i][j + k] === currentPlayer) {
        for (let index = 1; index < k; index += 1) {
          arr[i][j + index] = currentPlayer
        }
      }
    } else if (indI === 'none' && indJ === '-') {
      while (j - k >= 0 && arr[i][j - k] === opponent) {
        k += 1
      }
      if (j - k >= 0 && arr[i][j - k] === currentPlayer) {
        for (let index = 1; index < k; index += 1) {
          arr[i][j - index] = currentPlayer
        }
      }
    } else if (indI === '+' && indJ === '+') {
      while (i + k <= 7 && j + k <= 7 && arr[i + k][j + k] === opponent) {
        k += 1
      }
      if (i + k <= 7 && j + k <= 7 && arr[i + k][j + k] === currentPlayer) {
        for (let index = 1; index < k; index += 1) {
          arr[i + index][j + index] = currentPlayer
        }
      }
    } else if (indI === '+' && indJ === '-') {
      while (i + k <= 7 && j - k <= 7 && arr[i + k][j - k] === opponent) {
        k += 1
      }
      if (i + k <= 7 && j - k <= 7 && arr[i + k][j - k] === currentPlayer) {
        for (let index = 1; index < k; index += 1) {
          arr[i + index][j - index] = currentPlayer
        }
      }
    } else if (indI === '-' && indJ === '+') {
      while (i - k >= 0 && j + k <= 7 && arr[i - k][j + k] === opponent) {
        k += 1
      }
      if (i - k >= 0 && j + k <= 7 && arr[i - k][j + k] === currentPlayer) {
        for (let index = 1; index < k; index += 1) {
          arr[i - index][j + index] = currentPlayer
        }
      }
    } else if (indI === '-' && indJ === '-') {
      while (i - k >= 0 && j - k >= 0 && arr[i - k][j - k] === opponent) {
        k += 1
      }
      if (i - k >= 0 && j - k >= 0 && arr[i - k][j - k] === currentPlayer) {
        for (let index = 1; index < k; index += 1) {
          arr[i - index][j - index] = currentPlayer
        }
      }
    }

    let white = 0
    let black = 0
    arr.map((row) => {
      row.map((column) => {
        if (column === 'black') black += 1
        if (column === 'white') white += 1
      })
    })
    setWhiteScore(white)
    setBlackScore(black)
    return arr
  }

  const parseBoardToCheckAvailability = () => {
    const removeAllAvailable = board.map((row) => {
      return row.map((column) => {
        return column === 'available' ? 'no' : column
      })
    })
    const boardParsedTopToBottom = removeAllAvailable.map((row, i) => {
      if (i <= 5) {
        return row.map((column, j) => {
          if (column === currentPlayer || column === opponent || column === 'available' || removeAllAvailable[i + 1][j] !== opponent) {
            return column
          } else if (isAvailable(removeAllAvailable, i, j, '+', 'none')) {
            return 'available'
          }
          return column
        })
      }
      return row
    })
    const boardParsedBottomToTop = boardParsedTopToBottom.map((row, i) => {
      if (i >= 2) {
        return row.map((column, j) => {
          if (column === currentPlayer || column === opponent || column === 'available' || boardParsedTopToBottom[i - 1][j] !== opponent) {
            return column
          } else if (isAvailable(boardParsedTopToBottom, i, j, '-', 'none')) return 'available'
          return column
        })
      }
      return row
    })
    const boardParsedLeftToRight = boardParsedBottomToTop.map((row, i) => {
      return row.map((column, j) => {
        if (j <= 5) {
          if (column === currentPlayer || column === opponent || column === 'available' || boardParsedBottomToTop[i][j + 1] !== opponent) {
            return column
          } else if (isAvailable(boardParsedBottomToTop, i, j, 'none', '+')) return 'available'
        }
        return column
      })
    })
    const boardParsedRightToLeft = boardParsedLeftToRight.map((row, i) => {
      return row.map((column, j) => {
        if (j >= 2) {
          if (column === currentPlayer || column === opponent || column === 'available' || row[j - 1] !== opponent) {
            return column
          } else if (isAvailable(boardParsedLeftToRight, i, j, 'none', '-')) return 'available'
        }
        return column
      })
    })
    const boardParsedUpLeftToBottomRight = boardParsedRightToLeft.map((row, i) => {
      if (i <= 5) {
        return row.map((column, j) => {
          if (j <= 5) {
            if (column === currentPlayer || column === opponent || column === 'available' || boardParsedRightToLeft[i + 1][j + 1] === 'no' || boardParsedRightToLeft[i + 1][j + 1] === currentPlayer) {
              return column
            } else if (isAvailable(boardParsedRightToLeft, i, j, '+', '+')) return 'available'
          }
          return column
        })
      }
      return row
    })
    const boardParsedUpRightToBottomLeft = boardParsedUpLeftToBottomRight.map((row, i) => {
      if (i <= 5) {
        return row.map((column, j) => {
          if (j >= 2) {
            if (column === currentPlayer || column === opponent || column === 'available' || boardParsedUpLeftToBottomRight[i + 1][j - 1] === 'no' || boardParsedUpLeftToBottomRight[i + 1][j - 1] === currentPlayer) {
              return column
            } else if (isAvailable(boardParsedUpLeftToBottomRight, i, j, '+', '-')) return 'available'
          }
          return column
        })
      }
      return row
    })
    const boardParsedBottomLeftToUpRight = boardParsedUpRightToBottomLeft.map((row, i) => {
      if (i >= 2) {
        return row.map((column, j) => {
          if (j <= 5) {
            if (column === currentPlayer || column === opponent || column === 'available' || boardParsedUpRightToBottomLeft[i - 1][j + 1] === 'no' || boardParsedUpRightToBottomLeft[i - 1][j + 1] === currentPlayer) {
              return column
            } else if (isAvailable(boardParsedUpRightToBottomLeft, i, j, '-', '+')) return 'available'
          }
          return column
        })
      }
      return row
    })
    const boardParsedBottomRightToUpLeft = boardParsedBottomLeftToUpRight.map((row, i) => {
      if (i >= 2) {
        return row.map((column, j) => {
          if (j >= 2) {
            if (column === currentPlayer || column === opponent || column === 'available' || boardParsedBottomLeftToUpRight[i - 1][j - 1] === 'no' || boardParsedBottomLeftToUpRight[i - 1][j - 1] === currentPlayer) {
              return column
            } else if (isAvailable(boardParsedBottomLeftToUpRight, i, j, '-', '-')) return 'available'
          }
          return column
        })
      }
      return row
    })
    return setBoard(boardParsedBottomRightToUpLeft)
  }

  const parseBoardToTurnChip = (newBoard, i, j) => {
    if (i <= 5) {
      isTurning(newBoard, i, j, '+', 'none')
      if (j <= 5) {
        isTurning(newBoard, i, j, '+', '+')
      }
      if (j >= 2) {
        isTurning(newBoard, i, j, '+', '-')
      }
    }
    if (i >= 2) {
      isTurning(newBoard, i, j, '-', 'none')
      if (j <= 5) {
        isTurning(newBoard, i, j, '-', '+')
      }
      if (j >= 2) {
        isTurning(newBoard, i, j, '-', '-')
      }
    }
    if (j <= 5) {
      isTurning(newBoard, i, j, 'none', '+')
    }
    if (j >= 2) {
      isTurning(newBoard, i, j, 'none', '-')
    }

    setBoard(newBoard)
  }

  const parseBoardToCheckIfTheGameEnds = () => {
    let isTheBoardFilled = 0
    console.log(board)
    board.map((row) => {
      row.map((column) => {
        if (column === 'white' || column === 'black') {
          isTheBoardFilled += 1
        }
      })
    })
    let isThereBlackChip = 0
    let isThereWhiteChip = 0
    board.map((row) => {
      row.map((column) => {
        if (column === 'black') {
          isThereBlackChip += 1
        }
        if (column === 'white') {
          isThereWhiteChip += 1
        }
      })
    })
    if (isTheBoardFilled === 64 || isThereBlackChip === 0 || isThereWhiteChip === 0) {
      endGame()
    }
  }

  useEffect(() => {
    if (isInitialMount.current) {
      isInitialMount.current = false
    } else {
      parseBoardToCheckAvailability()
      board.length > 0 && parseBoardToCheckIfTheGameEnds()
      const game = {
        _id: id.id,
        board,
        player1: {
          pseudo: player1,
          color: 'black',
          score: blackScore
        },
        player2: {
          pseudo: player2,
          color: 'white',
          score: whiteScore
        },
        turn: currentPlayer,
        status,
        lastPass: lastPlayerPass
      }
      fetch('http://localhost:8888/updateGame', {
        method: 'PUT',
        headers: {
          'Content-type': 'application/json'
        },
        body: JSON.stringify(game)
      })
        .then((res) => res.json())
        .then((res) => {
          console.log('hiiiiiiiiii amjmajfamjf', res)
        })
    }
  }, [currentPlayer])

  return (
    <>
      <article className="board">
        <ul className="board__row">
          <li className="row-number"></li>
          <li className="row-number">1</li>
          <li className="row-number">2</li>
          <li className="row-number">3</li>
          <li className="row-number">4</li>
          <li className="row-number">5</li>
          <li className="row-number">6</li>
          <li className="row-number">7</li>
          <li className="row-number">8</li>
        </ul>
        {
          board.map((row, i) => (
            <ul key={`row${i}`} className="board__row">
              <li className="column-number">{ i + 1 }</li>
              {
                row.map((column, j) => {
                  if (column === 'available') {
                    return (
                      <li onClick={addChip(i, j)} className="board__square clickable" key={`row${i}column${j}`} id={`${i + 1}${j + 1}`} data-testid={`${i + 1}${j + 1}`}>
                        <Chip color={column} />
                      </li>
                    )
                  }
                  return (
                    <li className="board__square" key={`row${i}column${j}`} id={`${i + 1}${j + 1}`} data-testid={`${i + 1}${j + 1}`}>
                      <Chip color={column} />
                    </li>
                  )
                })
              }
            </ul>
          ))
        }
      </article>
    </>
  )
}

Board.propTypes = {
  currentPlayer: PropTypes.string.isRequired,
  opponent: PropTypes.string.isRequired,

  board: PropTypes.array.isRequired,

  setCurrentPlayer: PropTypes.func.isRequired,
  setOpponent: PropTypes.func.isRequired,
  setBoard: PropTypes.func.isRequired,
  setLastPlayerPass: PropTypes.func.isRequired,
  setWhiteScore: PropTypes.func.isRequired,
  setBlackScore: PropTypes.func.isRequired,
  endGame: PropTypes.func.isRequired
}

export default Board