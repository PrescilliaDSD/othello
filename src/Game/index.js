import React from 'react'

import Board from './Board'
import GameInfos from './GameInfos'

const Game = () => {
  return (
    <main>
      <h1>Othello</h1>
      <div className="play-area">
        <Board />
        <GameInfos />
      </div>
    </main>
  )
}

export default Game
