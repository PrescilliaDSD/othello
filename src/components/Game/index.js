/* eslint-disable react-hooks/exhaustive-deps */
import React, { useState, useEffect } from 'react'
import { useParams } from 'react-router-dom'

import Board from './Board'
import GameInfos from './GameInfos'

const Game = () => {
  const [currentPlayer, setCurrentPlayer] = useState('')
  const [opponent, setOpponent] = useState('')
  const [board, setBoard] = useState([])
  const [lastPlayerPass, setLastPlayerPass] = useState('')
  const [whiteScore, setWhiteScore] = useState(2)
  const [blackScore, setBlackScore] = useState(2)
  const [endMsg, setEndMsg] = useState('')
  const [player1, setPlayer1] = useState('')
  const [player2, setPlayer2] = useState('')
  const [status, setStatus] = useState('')

  const params = useParams()

  const endGame = () => {
    whiteScore > blackScore ? setEndMsg(`${player2} a gagné !`) : setEndMsg(`${player1} a gagné !`)
    setTimeout(() => {
      window.location.replace('http://localhost:9000/')
    }, 5000)
  }

  useEffect(() => {
    fetch('http://localhost:8888/game', {
      method: 'GET',
      headers: {
        'Content-type': 'application/json'
      }
    })
      .then((res) => res.json())
      .then((res) => {
        res.data.map((game) => {
          if (params.id === game._id) {
            setCurrentPlayer(game.turn)
            setOpponent(game.turn === 'black' ? 'white' : 'black')
            setBoard([...game.board])
            setLastPlayerPass(game.lastPass)
            setWhiteScore(game.player2.score)
            setBlackScore(game.player1.score)
            setPlayer1(game.player1.pseudo)
            setPlayer2(game.player2.pseudo)
            setStatus(game.status)
          }
          return game
        })
      })
  }, [])

  return (
    <main className="play-area">
      <Board endGame={endGame} whiteScore={whiteScore} blackScore={blackScore} setWhiteScore={setWhiteScore} setBlackScore={setBlackScore} setLastPlayerPass={setLastPlayerPass} board={board} setBoard={setBoard} currentPlayer={currentPlayer} setCurrentPlayer={setCurrentPlayer} opponent={opponent} setOpponent={setOpponent} player1={player1} player2={player2} status={status} lastPlayerPass={lastPlayerPass} />
      <GameInfos endMsg={endMsg} endGame={endGame} whiteScore={whiteScore} blackScore={blackScore} lastPlayerPass={lastPlayerPass} setLastPlayerPass={setLastPlayerPass} board={board} currentPlayer={currentPlayer} setCurrentPlayer={setCurrentPlayer} opponent={opponent} setOpponent={setOpponent} player1={player1} player2={player2} />
    </main>
  )
}

export default Game
