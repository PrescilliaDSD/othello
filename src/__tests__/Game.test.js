import React from 'react'
import renderer from 'react-test-renderer'
import { StaticRouter } from 'react-router-dom'

import Game from '../components/Game/index'

describe('Game component', () => {
  it('should render correctly', () => {
    const component = renderer.create(
      <StaticRouter params="123">
        <Game />
      </StaticRouter>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render a main tag', () => {
    const component = renderer.create(
      <StaticRouter params="123">
        <Game />
      </StaticRouter>
    )
    const tree = component.toJSON()
    expect(tree.type).toBe('main')
  })

  // it('should render a title in the main tag', () => {
  //   const component = renderer.create(
  //     <StaticRouter params="123">
  //       <Game />
  //     </StaticRouter>
  //   )
  //   const tree = component.toJSON()
  //   expect(tree.children[0].type).toBe('h1')
  // })

  // it('should render a title with content : "Othello"', () => {
  //   const tree = renderer
  //     .create(<Game />)
  //     .toJSON()
  //   expect(tree.children[0].children[0]).toBe('Othello')
  // })
})
