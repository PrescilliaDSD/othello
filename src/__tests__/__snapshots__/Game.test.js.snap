// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game component should render correctly 1`] = `
<main
  className="play-area"
>
  <article
    className="board"
  >
    <ul
      className="board__row"
    >
      <li
        className="row-number"
      />
      <li
        className="row-number"
      >
        1
      </li>
      <li
        className="row-number"
      >
        2
      </li>
      <li
        className="row-number"
      >
        3
      </li>
      <li
        className="row-number"
      >
        4
      </li>
      <li
        className="row-number"
      >
        5
      </li>
      <li
        className="row-number"
      >
        6
      </li>
      <li
        className="row-number"
      >
        7
      </li>
      <li
        className="row-number"
      >
        8
      </li>
    </ul>
  </article>
  <aside
    className="aside"
  >
    <div
      className="msg"
    >
      
    </div>
    <p>
      Score : 
    </p>
    <ul
      className="score__list"
    >
      <li
        className="score__list__item"
      >
        
         
        <div
          className="aside__chip black-chip chip"
        >
          <p>
            2
          </p>
        </div>
      </li>
      <li
        className="score__list__item"
      >
        
         
        <div
          className="aside__chip white-chip chip"
        >
          <p>
            2
          </p>
        </div>
      </li>
    </ul>
    <div
      className="turn"
    >
      C'est au tour de : 
      <div
        className="aside__chip -chip chip"
      />
    </div>
    <button
      className="pass-button"
      onClick={[Function]}
      type="button"
    >
      passer
    </button>
  </aside>
</main>
`;
